subprojects/core/src/main/java/org/gradle/execution/plan/DefaultExecutionPlan.java

```diff
@@ -645,6 +660,7 @@ public Node selectNext(WorkerLeaseRegistry.WorkerLease workerLease, ResourceLock
                 return node;
             }
         }
+        LOGGER.debug("No node could be selected, nodes ready: {}", foundReadyNode);
```

Comment:
Each of these logs add only a few nanoseconds, but the scheduler is a pretty hot place, too. I wonder how many times do we execute these log statements in a large build.

Reply:
We execute the log statements:
- each time a node completes
- the first time no ready node is found
- every time a lock changes and there are ready nodes

The lock changing does not happen to often, and most times no nodes are ready for execution. So logging (or checking if debug is enabled) should not happen too often.